arch/arm64/src/qemu/qemu_serial.c

```diff
@@ -0,0 +1,856 @@
+#define PL011_CR_RXE                    BIT(9)  /* receive enable */
+#define PL011_CR_DTR                    BIT(10) /* data transmit ready */
+#define PL011_CR_RTS                    BIT(11) /* request to send */
+#define PL011_CR_Out1                   BIT(12)
+#define PL011_CR_Out2                   BIT(13)
+#define PL011_CR_RTSEn                  BIT(14) /* RTS hw flow control enable
+                                                 */
+#define PL011_CR_CTSEn                  BIT(15) /* CTS hw flow control enable
+                                                 */
+
+/* PL011 Interrupt Fifo Level Select Register */
+#define PL011_IFLS_TXIFLSEL_SHIFT       0   /* bits 2:0 */
+#define PL011_IFLS_TXIFLSEL_WIDTH       3
+#define PL011_IFLS_RXIFLSEL_SHIFT       3   /* bits 5:3 */
+#define PL011_IFLS_RXIFLSEL_WIDTH       3
+
+/* PL011 Interrupt Mask Set/Clear Register */
+#define PL011_IMSC_RIMIM                BIT(0)  /* RTR modem interrupt mask */
+#define PL011_IMSC_CTSMIM               BIT(1)  /* CTS modem interrupt mask */
+#define PL011_IMSC_DCDMIM               BIT(2)  /* DCD modem interrupt mask */
+#define PL011_IMSC_DSRMIM               BIT(3)  /* DSR modem interrupt mask */
+#define PL011_IMSC_RXIM                 BIT(4)  /* receive interrupt mask */
+#define PL011_IMSC_TXIM                 BIT(5)  /* transmit interrupt mask */
+#define PL011_IMSC_RTIM                 BIT(6)  /* receive timeout interrupt
+                                                 * mask */
+#define PL011_IMSC_FEIM                 BIT(7)  /* framing error interrupt
+                                                 * mask */
+#define PL011_IMSC_PEIM                 BIT(8)  /* parity error interrupt mask
+                                                 */
+#define PL011_IMSC_BEIM                 BIT(9)  /* break error interrupt mask
+                                                 */
+#define PL011_IMSC_OEIM                 BIT(10) /* overrun error interrupt
+                                                 * mask */
+
+#define PL011_IMSC_ERROR_MASK           (PL011_IMSC_FEIM |                   \
+                                         PL011_IMSC_PEIM | PL011_IMSC_BEIM | \
+                                         PL011_IMSC_OEIM)
+
+#define PL011_IMSC_MASK_ALL             (PL011_IMSC_OEIM | PL011_IMSC_BEIM | \
+                                         PL011_IMSC_PEIM | PL011_IMSC_FEIM | \
+                                         PL011_IMSC_RIMIM |                  \
+                                         PL011_IMSC_CTSMIM |                 \
+                                         PL011_IMSC_DCDMIM |                 \
+                                         PL011_IMSC_DSRMIM |                 \
+                                         PL011_IMSC_RXIM | PL011_IMSC_TXIM | \
+                                         PL011_IMSC_RTIM)
+
+static void pl011_enable(const struct pl011_uart_port_s *sport)
```

Comment:
Could you please add Defines, Types, Public/Private Functions sections? Here and in other files